### PR TITLE
Update expand_source_files( )

### PR DESCRIPTION
**Pull request by ritwik12**
Solves #35 as requested in [111](https://github.com/google/pytype/issues/111), Allow files with all extensions irrespective of ".py"

### CODE REVIEW

File: importlab/utils.py [hunk 1]
```diff
@@ -66,22 +66,21 @@ def expand_source_files(filenames, cwd=None):
...
-        A set of full paths to .py files
+        A set of full paths to files
     """
     out = []
     for f in expand_paths(filenames, cwd):
         if os.path.isdir(f):
-            # If we have a directory, collect all the .py files within it.
-            out += collect_files(f, ".py")
+            # If we have a directory, collect all the files within it.
+            out += collect_files(f, ".*")
```
Comment (martindemello) [inline importlab/utils.py]: this is not going to work; both `collect_files` and this function are doing a literal string match on the extension. you'll need to change `collect_files` to something like
````
if extension is None:
  out += [os.path.join(root, f) for f in files]
else:
  out += [os.path.join(root, f) for f in files if f.endswith(extension)]
```

and in this function simply remove the `if f.endswith...` block

Comment (ritwik12) [reply importlab/utils.py]: @martindemello Thanks for helping. I missed that

Comment (ritwik12) [reply importlab/utils.py]: @martindemello Thanks for your help

File: importlab/utils.py [hunk 2]
```diff
@@ -66,23 +69,21 @@ def expand_source_files(filenames, cwd=None):
...
```
Comment (martindemello) [inline importlab/utils.py]: this line needs to be collect_files(f, None) now

**Comment by rchen152** [comment]
By the way, the changes in this PR probably shouldn't be released to PyPI until https://github.com/google/pytype/pull/204 has been released (or we should update pytype's setup.py to pin to the current version of importlab), since I think it would cause pytype's unreadable input checking to complain about all sorts of non-Python files.

**Comment by martindemello** [comment]
also, could you add some tests for your change? the best place would be in tests/graph.py; you can look at these two tests for how to do it:

https://github.com/google/importlab/blob/master/tests/test_graph.py#L145

File: importlab/utils.py (hunk 2 above)
Comment (ritwik12) [reply importlab/utils.py]: Done

**Comment by ritwik12** [comment]
@martindemello Sure, I'll add test cases.

**Comment by ritwik12** [comment]
@martindemello WIll removing `test_unreadable_direct_source` and `test_unreadable_direct_source` work? or We need to write one more function for `test_readable`? 
I am not able to understand few things in the tests.

**Comment by martindemello** [comment]
@ritwik12 the existing tests should not be broken by this change; you will need to add another test or two to test that we handle both readable and unreadable files without a .py extension.

the tests work by setting up a virtual file system as a dict of

```
# Deps = { file : ([resolved deps], [broken deps], {dep_file:provenance}) }
```

(look at `SIMPLE_DEPS` for instance, and see if you can make out what each entry in there is doing; please do ask questions if something is hard to follow because we would like the tests to be readable)

you will need to add a new dict, e.g. `NON_PY_FILES`, which copies `SIMPLE_DEPS` and adds another couple of entries for files with no extension, one with deps and one without deps. Then, when constructing a `FakeImportGraph` in the test, you pass it a set of unreadable files, so you can mark the file without deps as unreadable, and see that it gets handled correctly.

**Comment by ritwik12** [comment]
@martindemello Thanks for your explaination
